### ISSUE TITLE
Decorations from multiple extensions interfering with each other

### ISSUE THREAD

**Issue by skdeng** [microsoft/vscode#91185]
Version: 1.43.0-insider (user setup)
Commit: e6a45f4242ebddb7aa9a229f85555e8a3bd987e2
Date: 2020-02-21T05:42:20.536Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.17763

Decorations from multiple extensions seems to be interfering with each other. In the example below, I create a decoration that append `text` in red at the cursor location. However, with the [GitLens ](https://github.com/eamodio/vscode-gitlens) extension enabled side-by-side, the decoration is being moved by the GitLens extension.

**Code:**
```
import * as vscode from "vscode";

export function activate(context: vscode.ExtensionContext) {
  const decorationType = vscode.window.createTextEditorDecorationType({});
  const updateDecoration = () => {
    const activeEditor = vscode.window.activeTextEditor;
    if (activeEditor) {
      const decorationOptions = [
        {
          range: new vscode.Range(
            activeEditor.selection.active,
            activeEditor.selection.active
          ),
          renderOptions: {
            after: {
              contentText: "text",
              color: "red"
            }
          }
        }
      ];
      activeEditor.setDecorations(decorationType, decorationOptions);
    }
  };

  vscode.workspace.onDidChangeTextDocument(event => {
    updateDecoration();
  });
}

export function deactivate() {}
```

**Behavior:**
![vsce](https://user-images.githubusercontent.com/3770270/75074479-a6583f80-54b0-11ea-878b-4f0dbce703b4.gif)

Does this issue occur when all extensions are disabled?: No

**Comment by usernamehw** [comment]
Duplicate of #33852
